src/localization/plugin.ts

```diff
@@ -1,151 +0,0 @@
-/* eslint-disable func-names */
-/* eslint-disable no-param-reassign */
-/* eslint-disable no-restricted-syntax */
-import mongoose from 'mongoose';
-import sanitizeFallbackLocale from './sanitizeFallbackLocale';
-import formatRefPathLocales from './formatRefPathLocales';
-
-export default function localizationPlugin(schema: any, options): void {
-  if (!options || !options.locales || !Array.isArray(options.locales) || !options.locales.length) {
-    throw new mongoose.Error('Required locales array is missing');
-  }
-
-  schema.eachPath((path, schemaType) => {
-    if (schemaType.schema) { // propagate plugin initialization for sub-documents schemas
-      schemaType.schema.plugin(localizationPlugin, options);
-    }
-
-    if (!schemaType.options.localized && !(schemaType.schema && schemaType.schema.options.localized)) {
-      return;
-    }
-
-    if (schemaType.options.unique) {
-      schemaType.options.sparse = true;
```

Comment:
I think we'll need to set sparse to true in our buildSchema when a field is both unique && localized

Reply:
Good call!